Issue 21/cache refresh

tcnichol opened the pull request:
refresh cache script

tcnichol commented:
This refresh script should work by itself now. It is ready for testing/review.

max-zilla commented:
I'm pushing a few updates. Going forward, please remove test files/print statements before creating a PR because we don't need to keep those in production. removing some now.

max-zilla commented on Dockerfile:
```diff
@@ -28,7 +28,8 @@ RUN apt-get update -qq \
       && rm -rf /srv/shiny-server/sample-apps
       
 COPY . /srv/shiny-server/
-COPY shiny-server.conf /etc/shiny-server/
```
I'm adding this back in - we can't remove this or it breaks the app.

max-zilla commented on cache-refresh.R:
```diff
@@ -121,7 +118,12 @@ get_data_for_subexp <- function(subexp, exp_name) {
+  file.create("/srv/shiny-server/cache/cache.RData.temp")
+  save(full_cache_data, file = "/srv/shiny-server/cache/cache.RData.temp", compress = FALSE)
+  # copy file to cache.RData
+  file.copy("/srv/shiny-server/cache/cache.RData.temp","/srv/shiny-server/cache/cache.RData",overwrite=TRUE)
```
changing this to file.rename, we dont need to keep the temp copy for any reason

max-zilla commented:
We need to make sure that when cache-server reloads the cache file, that the app.R script correctly uses the new file instead of using the already-loaded data.

tcnichol commented:
@robkooper 

i refactored the loading of the cache data and added a check for whether the cache file has been modified. not sure if it's correct, so please check and comment.
